tests/operator_test.go

```diff
@@ -1467,6 +1467,126 @@ spec:
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vc.Spec.Template.Spec.ImagePullSecrets).To(Equal(imagePullSecrets))
+
+			va, err := virtClient.AppsV1().Deployments(originalKv.Namespace).Get(context.Background(), "virt-api", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(va.Spec.Template.Spec.ImagePullSecrets).To(Equal(imagePullSecrets))
+
+			vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), "virt-handler", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vh.Spec.Template.Spec.ImagePullSecrets).To(Equal(imagePullSecrets))
+
+			if len(imagePullSecrets) == 0 {
+				Expect(vh.Spec.Template.Spec.Containers).To(HaveLen(1))
+			} else {
+				Expect(vh.Spec.Template.Spec.Containers).To(HaveLen(2))
+				Expect(vh.Spec.Template.Spec.Containers[1].Name).To(Equal("virt-launcher-image-holder"))
+			}
+		}
+
+		It("test empty imagePullSecrets", func() {
+
+			By("Check that KubeVirt CR has empty imagePullSecrets")
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(kv.Spec.ImagePullSecrets).To(HaveLen(0))
+
+			By("Ensuring that all virt components have empty image pull secrets")
+			checkVirtComponents(nil)
+
+			By("Starting a VMI")
+			vmi := tests.NewRandomVMI()
+			vmi, err = virtClient.VirtualMachineInstance(util2.NamespaceTestDefault).Create(vmi)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("Getting virt-launcher")
+			uid := vmi.GetObjectMeta().GetUID()
+			labelSelector := fmt.Sprintf(v1.CreatedByLabel + "=" + string(uid))
+			pods, err := virtClient.CoreV1().Pods(util2.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
+			Expect(err).NotTo(HaveOccurred(), "Should list pods")
+			Expect(pods.Items).To(HaveLen(1))
+			Expect(pods.Items[0].Spec.ImagePullSecrets).To(HaveLen(0))
+
+		})
+
+		It("test applying imagePullSecrets", func() {
+
+			const imagePullSecretName = "testmyregistrykey"
+			var imagePullSecrets = []k8sv1.LocalObjectReference{{Name: imagePullSecretName}}
+
+			By("Delete existing image pull secret")
+			_ = virtClient.CoreV1().Secrets(originalKv.Namespace).Delete(context.Background(), imagePullSecretName, metav1.DeleteOptions{})
+			By("Create image pull secret")
+			_, err := virtClient.CoreV1().Secrets(originalKv.Namespace).Create(context.Background(), &k8sv1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imagePullSecretName,
+					Namespace: originalKv.Namespace,
+				},
+				Type: k8sv1.SecretTypeDockerConfigJson,
+				Data: map[string][]byte{
+					".dockerconfigjson": []byte("UmVhbGx5IHJlYWxseSByZWVlZWVlZWVlZWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWxsbGxsbGxsbGxsbGxsbGxsbGxsbGxsbGxsbGxsbGx5eXl5eXl5eXl5eXl5eXl5eXl5eSBsbGxsbGxsbGxsbGxsbG9vb29vb29vb29vb29vb29vb29vb29vb29vb25ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubmdnZ2dnZ2dnZ2dnZ2dnZ2dnZ2cgYXV0aCBrZXlzCg=="),
```

Comment:
Looks to me like it does not expect it base16 encoded whne looking at the e2e test:

```
Message: "Secret \"testmyregistrykey\" is invalid: data[.dockerconfigjson]: Invalid value: \"<secret contents redacted>\": invalid character 'U' looking for beginning of value",
```